app/src/main/java/org/dhis2/data/forms/dataentry/EnrollmentRepository.kt

```diff
@@ -230,6 +245,8 @@ class EnrollmentRepository(
 
         return if (warning != null) {
             fieldViewModel.withWarning(warning)
+        } else if (!error.isNullOrEmpty()) {
```

Comment:
I would check error first and then warning

Reply:
ok